sql/src/main/java/io/crate/analyze/InsertFromValuesAnalyzer.java

```diff
@@ -607,4 +609,19 @@ private void addGeneratedPartitionedColumnValue(ColumnIdent columnIdent,
                 givenValue, generatedValue));
         }
     }
+
+    static void verifyOnConflictTargets(List<String> constraintColumns, List<ColumnIdent> pkColumnIdents) {
+        if (constraintColumns.size() != pkColumnIdents.size()) {
+            throw new IllegalArgumentException(
+                String.format("Number of conflict targets (%s) did not match the number of primary key columns (%s)",
+                    constraintColumns, pkColumnIdents));
+        }
+        for (ColumnIdent pkColIdent : pkColumnIdents) {
```

Comment:
doesn't the order also matter?

Reply:
The order doesn't matter in PostgreSQL, what matters is that the constraint can be inferred.